internal/apiserver/route_get_txn_blockchainevents.go

```diff
@@ -0,0 +1,45 @@
+// Copyright © 2022 Kaleido, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package apiserver
+
+import (
+	"net/http"
+
+	"github.com/hyperledger/firefly/internal/config"
+	"github.com/hyperledger/firefly/internal/i18n"
+	"github.com/hyperledger/firefly/internal/oapispec"
+	"github.com/hyperledger/firefly/pkg/fftypes"
+)
+
+var getTxnBlockchainEvents = &oapispec.Route{
+	Name:   "getTxnBlockchainEvents",
+	Path:   "namespaces/{ns}/transactions/{txnid}/blockchainevents",
+	Method: http.MethodGet,
+	PathParams: []*oapispec.PathParam{
+		{Name: "ns", ExampleFromConf: config.NamespacesDefault, Description: i18n.MsgTBD},
+		{Name: "txnid", Description: i18n.MsgTBD},
+	},
+	QueryParams:     nil,
+	FilterFactory:   nil,
+	Description:     i18n.MsgTBD,
+	JSONInputValue:  nil,
+	JSONOutputValue: func() interface{} { return &[]*fftypes.BlockchainEvent{} },
```

Comment:
I copy-pasted this, but your comment prompted me to do a quick investigation... it actually looks like the Swagger generation for the output schema is only correct if you _do_ include the `&`. So for this route it correctly indicates that it will return an _array_ of BlockchainEvent. But for other routes that don't return a pointer, it's a bit broken - such as for `/transactions`, which returns `[]*fftypes.Transaction`, the Swagger indicates it will return a single Transaction object.

I take this to mean we should add the `&` everywhere on array returns, or investigate why the Swagger generation behaves this way.

Reply:
Thanks for digging! Will merge based on this and raise an issue

Reply:
Thanks, I see #449